lock-api/src/main/java/com/palantir/lock/LockClient.java

```diff
@@ -66,6 +68,11 @@ public static LockClient of(String clientId) {
         Preconditions.checkArgument(!clientId.equals(INTERNAL_LOCK_GRANT_CLIENT_ID));
         return new LockClient(clientId);
     }
+    
+    @JsonCreator
+    public static LockClient valueOf(String clientId) {
```

Comment:
@rjullman Seems like `atlasdb-server/src/test/java/com/palantir/server/LockRemotingTest.java` should be exercising `lockAndGetHeldLocks` and `lockWithFullLockResponse` as well, and `com.palantir.lock.LockResponse` also seems to be missing Jackson annotations.